src/frameworks/next/index.ts

```diff
@@ -288,7 +287,9 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
   ]);
 
   const middlewareMatcherRegexes = Object.values(middlewareManifest.middleware)
-    .map((it) => it.matchers)
+    .map((page: MiddlewareManifest["middleware"]["page"]) =>
```

Comment:
Test this branch? (bonus points illustrating a test -case a previously broken case and showing it works now.)